pulsar-common/src/main/java/org/apache/pulsar/common/nar/NarClassLoader.java

```diff
@@ -283,4 +304,32 @@ protected String findLibrary(final String libname) {
     public String toString() {
         return NarClassLoader.class.getName() + "[" + narWorkingDirectory.getPath() + "]";
     }
+
+    @Override
+    public void close() throws IOException {
+        if (this.release() == 0) {
+            CACHED_CLASS_LOADER.compute(classLoaderCacheKey, (k, narClassLoader) -> {
```

Comment:
I think that this should be `remove`

Reply:
Return null in `compute` block also can remove the classLoader from the cache.